core/mock/state.go

```diff
@@ -36,133 +36,153 @@ type PrivateOperatorState struct {
 	PrivateOperators map[core.OperatorID]*PrivateOperatorInfo
 }
 
-func makeOperatorId(id int) core.OperatorID {
-	data := [32]byte{}
-	copy(data[:], []byte(fmt.Sprintf("%d", id)))
+func MakeOperatorId(id int) core.OperatorID {
+	data := [32]byte{uint8(id)}
 	return data
 }
 
-func NewChainDataMock(stakes []int) (*ChainDataMock, error) {
-	numOperators := uint(len(stakes))
-
-	keyPairs := make([]*core.KeyPair, numOperators)
-	for ind := core.OperatorIndex(0); ind < numOperators; ind++ {
-		keyPair, err := core.GenRandomBlsKeys()
-		if err != nil {
-			return nil, err
+func NewChainDataMock(stakes map[core.QuorumID]map[core.OperatorID]int) (*ChainDataMock, error) {
+	numOperators := 0
+	keyPairs := make(map[core.OperatorID]*core.KeyPair)
+	for _, oprStakes := range stakes {
+		if len(oprStakes) > 255 {
+			return nil, errors.New("too many operators")
+		}
+		if len(oprStakes) > numOperators {
```

Comment:
The operator set in each quorum are not exclusive. Some operators overlap across multiple quorums as explained in L73

Reply:
I agree with @jianoaix. I think the write way to do this would be to create a map `uniqueOperators map[core.OperatorID]struct{}`, add all of the operators to it, and then use its length.